[TEST] failures in integ-test-zip

I am not able to run `gradle check` locally on the current master (315319b76325e6385f4c689329f875bf0a43ca6e) as all tests in `integ-test-zip` fail. The log is full of
```
<testcase classname="org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT" name="test {p0=snapshot.get/10_basic/Get missing snapshot info throws an exception}" time="0.007">
      <failure message="" type="java.lang.AssertionError">java.lang.AssertionError
	at org.elasticsearch.test.rest.ESRestTestCase.initClient(ESRestTestCase.java:99)
	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1713)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:941)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:957)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:916)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:802)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:852)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:745)
</failure>
      <error message="" type="java.lang.NullPointerException">java.lang.NullPointerException
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:203)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:131)
	at sun.reflect.GeneratedMethodAccessor7.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1713)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:965)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:916)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:802)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:852)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:745)
</error>
      <error message="" type="java.lang.NullPointerException">java.lang.NullPointerException
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.afterIfFailed(ESClientYamlSuiteTestCase.java:150)
	at org.elasticsearch.test.ESTestCase$1.afterAlways(ESTestCase.java:215)
	at com.carrotsearch.randomizedtesting.rules.TestRuleAdapter$1.afterAlways(TestRuleAdapter.java:31)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:43)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:916)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:802)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:852)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:745)
</error>
```
Running a single test 
```
gradle :distribution:integ-test-zip:integTestRunner -Dtests.seed=A52D8951AA029B4D -Dtests.class=org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT -Dtests.method="test {p0=snapshot.get/10_basic/Get missing snapshot info throws an exception}" -Dtests.security.manager=true -Dtests.jvms=16 -Dtests.locale=ar-SD -Dtests.timezone=Asia/Khandyga
``` 
results in : 
```
<testcase classname="org.elasticsearch.test.rest.IntegTestZipClientYamlTestSuiteIT" name="initializationError" time="0.006">
      <error message="org/elasticsearch/client/ResponseException" type="java.lang.NoClassDefFoundError">java.lang.NoClassDefFoundError: org/elasticsearch/client/ResponseException
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.getDeclaredMethods(Class.java:1975)
	at com.carrotsearch.randomizedtesting.ClassModel$3.members(ClassModel.java:215)
	at com.carrotsearch.randomizedtesting.ClassModel$3.members(ClassModel.java:212)
	at com.carrotsearch.randomizedtesting.ClassModel$ModelBuilder.build(ClassModel.java:85)
	at com.carrotsearch.randomizedtesting.ClassModel.methodsModel(ClassModel.java:212)
	at com.carrotsearch.randomizedtesting.ClassModel.&lt;init&gt;(ClassModel.java:207)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.&lt;init&gt;(RandomizedRunner.java:329)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
	at com.carrotsearch.ant.tasks.junit4.slave.SlaveMain.execute(SlaveMain.java:246)
	at com.carrotsearch.ant.tasks.junit4.slave.SlaveMain.main(SlaveMain.java:368)
	at com.carrotsearch.ant.tasks.junit4.slave.SlaveMainSafe.main(SlaveMainSafe.java:13)
Caused by: java.lang.ClassNotFoundException: org.elasticsearch.client.ResponseException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 22 more
</error>
```
Running `bisect` on the master results in `e816ef89a29337d88bb19d89e322d451efb3b6fb is the first bad commit`

--- a/client/rest/build.gradle
+++ b/client/rest/build.gradle
@@ -84,29 +84,19 @@ task shadeDeps(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar)
   }
 }
 
-// builds the actual shaded jar that is being distributed
-shadowJar {
-  configurations = [project.configurations.shade]
-  classifier = null
-  relocate 'org.apache', 'org.elasticsearch.client'
-}
-
 jar {
-  // move the default jar to another classifier so the jar with the shaded dependencies can have the empty classifier
-  classifier = 'nodeps'
+  from zipTree(shadeDeps.outputs.files.singleFile)
+  dependsOn shadeDeps
 }
 
-// Mangle this build so it outputs the correct artifacts
-// readds the shadowJar output (the actual uber jar) but only from the cli
+// remove the deps jar from the classpath to avoid jarHell
 if (isIdea == false && isEclipse == false) {
-  // removes the -nodeps jar
-  configurations.runtime.artifacts.removeAll { it.archiveTask.is jar }
-
-  // removes the runtime configuration inheritance from compile
+  // runtimeElements does not exist in 3.3
+  configurations.maybeCreate('runtimeElements')
+  // 3.3 cleanup to remove the deps jar from the classpath
   configurations.runtime.extendsFrom -= [configurations.compile]
-  artifacts {
-    runtime project.tasks.shadowJar
-  }
+  // 3.5+ cleanup to remove the deps jar from the classpath
+  configurations.runtimeElements.extendsFrom = []
 }
 
 // adds a dependency to compile, so the -deps jar is built first

--- a/client/rest/build.gradle
+++ b/client/rest/build.gradle
@@ -21,22 +21,18 @@ import org.elasticsearch.gradle.precommit.PrecommitTasks
 
 /**
  * The rest client is a shaded jar. It contains the source of the rest client, as well as all the dependencies,
- * shaded to the `org.elasticsearch.client` package. 3 artifacts come out of this build process. The shading process only
- * modifies the imports and class names and locations. It does not do any processing on the files. The classes used to
- * interact with the rest client are no different from the dependencies in the shade configuration, besides in name.
+ * shaded to the `org.elasticsearch.client` package. 2 artifacts come out of this build process. The shading process
+ * only modifies the imports and class names and locations. It does not do any processing on the files. The classes used
+ * to interact with the rest client are no different from the dependencies in the shade configuration, besides in name.
  *
  * IDEs do not like removing artifacts and changing configurations on the fly, so the bits that make the build use the
- * actual shaded jar (3) are only executed on the cli. Tests run in an IDE rely on the deps (1) and nodeps (2) jars.
+ * actual shaded jar (2) are only executed on the cli. Tests run in an IDE rely on the deps (1) jar.
  *
- * 1) A jar that contains *only* the `org.elasticsearch.client` dependencies. This is a jar that is built before the src
- *    is compiled. This jar is only used by the rest client so its compilable. There exists a chicken-egg situation where
- *    the src needs compilation and depends on `org.elasticsearch.client` classes, so an intermediary jar needs to exist
- *    to satisfy the compile. The `deps` classifier is added to this jar.
- * 2) The original src/ contents, using the `org.elasticsearch.client` packages, but without the `org.elasticsearch.client`
- *    code also in the jar. The reason this jar was not overwritten was because gradle knows what its inputs and outputs
- *    are for a given task and we should not alter that by overwriting the original jar. What we do is add a `nodeps`
- *    classifier to it instead.
- * 3) The *actual* jar that will be used by clients. This has no classifier, contains the rest client src and
+ * 1) A jar that contains *only* the `org.elasticsearch.client` shaded dependencies. This is a jar that is built before
+ *    the src is compiled. This jar is only used by the rest client so will compile. There exists a chicken-egg
+ *    situation where the src needs compilation and depends on `org.elasticsearch.client` shaded classes, so an
+ *    intermediary jar needs to exist to satisfy the compile. The `deps` classifier is added to this jar.
+ * 2) The *actual* jar that will be used by clients. This has no classifier, contains the rest client src and
  *    `org.elasticsearch.client`. This jar is the only actual output artifact of this job.
  */
 plugins {

--- a/client/rest/build.gradle
+++ b/client/rest/build.gradle
@@ -87,12 +87,12 @@ jar {
 
 // remove the deps jar from the classpath to avoid jarHell
 if (isIdea == false && isEclipse == false) {
-  // runtimeElements does not exist in 3.3
-  configurations.maybeCreate('runtimeElements')
-  // 3.3 cleanup to remove the deps jar from the classpath
-  configurations.runtime.extendsFrom -= [configurations.compile]
-  // 3.5+ cleanup to remove the deps jar from the classpath
-  configurations.runtimeElements.extendsFrom = []
+  // cleanup to remove the deps jar from the classpath
+  if (gradle.gradleVersion == "3.3") {
+    configurations.runtime.extendsFrom -= [configurations.compile]
+  } else if (gradle.gradleVersion > "3.3") {
+    configurations.runtimeElements.extendsFrom = []
+  }
 }
 
 // adds a dependency to compile, so the -deps jar is built first

--- a/core/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy
+++ b/core/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy
@@ -77,14 +77,6 @@ grant codeBase "${codebase.elasticsearch-rest-client-6.0.0-beta1-SNAPSHOT-deps.j
   permission java.net.SocketPermission "*", "connect";
 };
 
-// IDEs need this because they do not play nicely with removing artifacts on projects,
-// so we keep it in here for IDE test support
-grant codeBase "${codebase.elasticsearch-rest-client-6.0.0-beta1-SNAPSHOT-nodeps.jar}" {
-  // rest makes socket connections for rest tests
-  permission java.net.SocketPermission "*", "connect";
-};
-
-
 grant codeBase "${codebase.httpcore-nio-4.4.5.jar}" {
   // httpcore makes socket connections for rest tests
   permission java.net.SocketPermission "*", "connect";

